### PR TITLE
Modernize bhDateEditor component

### LINKED ISSUE #1960: refactor: modernize bhDateEditor
The [bhDateEditor](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/components/bhDateEditor.js) component is one of the core components used in every single page.  It was one of the first components completed, and the design has aged.  Here are some refactors to bring it up to date with standards:
 1. Use the translate directive instead of the translate filter.
 2. Initialize label in the `$onInit` method.
 3. BREAKING CHANGE: remove the two-way bound data and use an `onChange()` to update the parent controller's date input.
 4. Use the `$$parent` form property to validate the date input.
 5. Use font-awesome instead of glyphicons.

### PR DESCRIPTION
**Pull request by jniles**
This commit modernizes the bhDateEditor component by using a one-way binding and callbacks to trigger changes on the component's border.  The `onChange()` callback has been added to propagate the changes to the parent controller.

The component signature looks like this:
```html
<bh-date-editor date-value="Controller.someDateValue" on-change="Controller.onDateChange(date)">
</bh-date-editor>
```

In addition to these changes, a few unit tests have been written.  Finally, ESLint was run on most client controllers that I touched.

Closes #1960.

### CODE REVIEW

File: client/src/js/components/bhDateEditor.js [hunk 1]
```diff
@@ -1,19 +1,20 @@
...
-    disabled          : '<', // one-way binding
-    dateFormat        : '@', // bind text
-    label             : '@', // bind text
-  },
-});
+  .component('bhDateEditor', {
+    templateUrl : 'modules/templates/bhDateEditor.tmpl.html',
+    controller : bhDateEditorController,
+    bindings : {
+      dateValue : '<', // one-way binding
```
Comment (mbayopanda) [inline client/src/js/components/bhDateEditor.js:6]: 👍

File: client/src/js/components/bhDateEditor.js [hunk 2]
```diff
@@ -25,6 +26,7 @@ dateEditorController.$inject = ['bhConstants'];
  * @example
  * <bh-date-editor
  *  date-value="Ctrl.date"
+ *  on-change="Ctrl.onDateChange(date)"
```
Comment (mbayopanda) [inline client/src/js/components/bhDateEditor.js:29]: 👍

**Review by mbayopanda — APPROVED** [review]
~~LGMT~~  LGTM. 💃

**Comment by mbayopanda** [comment]
@jniles could you fix these conflicts ?

**Comment by jniles** [comment]
Fixed!  Let's see if tests pass...

**Pull request merged**